publish-new-version.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+
+# publish-new-version.sh 
+# EXAMPLE USAGE: publish-new-version.sh 0.1.1
+#
+# This script accepts a desired new crate version as an argument and updates 
+# the Cargo.toml version of the repository to the given version, builds the 
+# crate, commits the changes, publishes the crate, and then pushes the crate 
+# to github.
+#
+# The script performs some types of sanity checking, but in general you should
+# read and understand this script before running it.
+#
+# This script assumes that your git upstream is called "origin".
+
+set -eu
+
+command_exists() {
+    command -v "$1" >/dev/null 2>&1 || ( echo "Command \`$1\` isn't available. Please install before continuing."; exit 1 )
+}
+
+update_cargo_toml_version() {
+  crate_version=$1
+  sed -i "s/^version =.*/version = \"$crate_version\"/g" Cargo.toml
+}
+
+cargo_build_crate() {
+  cargo build
```

Comment:
Should this have `--release` with it?

Reply:
Done.

Added explanation for why we're doing this as well.